### ISSUE TITLE
[Enhancement] - `#[test_case]` parsing multithreading

### ISSUE THREAD

**Issue by nannafudge** [nannafudge/sith#18]
**This is to remain low priority until profiling, or performance proves to be an issue in large projects/test suites (if at all). The former implies most primary/secondary features are out of the way, which gives space for tertiary optimization**

###### *Which functionality/feature of Sith does this enhancement pertain to?*

`#[test_case]`

---

### Rationale

> *Reasoning behind the enhancement, i.e. why/what it would enable*

When parsing `#[test_case]` attributes, there may (or likely will) be several present on the same function, each defining their own parameters. I.e:

```rust
#[test_case(bar)]
#[test_case(baz)]
#[test_case(qux)]
fn foo() {
    ...
}
```

In this case, when being parsed, a `#[test_case]` must check for any additional `#[test_case]` attributes (besides itself) on the target fn definition, ensuring each is parsed and has its mutators executed against a fresh copy of the function.

In implementation, we currently handle this by:

- [iterate through the remaining attributes on the target fn signature](https://github.com/nannafudge/sith/blob/main/src/core/test_case/mod.rs#L122)
- [if target attribute name ends with `test_case`, pop it from the fn attributes and parse it](https://github.com/nannafudge/sith/blob/main/src/core/test_case/mod.rs#L122)
- [once all have been parsed, execute each's respective mutators against a fresh copy of the target function](https://github.com/nannafudge/sith/blob/main/src/core/test_case/mod.rs#L138)

As each `test_case` is discrete/separate from one another, this process is a prime candidate for a lazy/work-stealing thread pool. Parsing/modifying syntax trees is quite process-intensive, especially with the amount of cloning required ala. syn. I imagine there'd be a great performance uplift - although awaiting profiling.

---

### Implementation Details

May depend on #15, if it utilizes the same underlying implementation. Wholly depends on approach there regarding applicability - may be best to settle on Rayon for both, although that adds more overhead in compilation.

Otherwise, if a thread pool is created, a work stealer could effectively be implemented atop a lightweight SPMC library. Alternatively, can roll our own SPMC (conceptually simple so long as you follow da rules, ala CAS) with atomics.

**Considering #3 will need to expand sub test-cases, this may also be a good considering for that too**

> *Any additional information that may help in understanding/implementing the enhancement - i.e. ideas, direction based on current codebase & thoughts*

- [x] **Can this be done by yourself, or does it require third-party assistance**

- [ ] **Has this already been implemented by yourself (please link PR below if so)**

---

### Additional Comments

> *Additional comments/notes/screenshots regarding the enhancement*
